### ISSUE TITLE
Bug on some static variables

### ISSUE THREAD

**Issue by Rolige** [sirbrillig/phpcs-variable-analysis#79]
Hi,

I think this bug was inserted in the last release, the problem occurs with some variables defined as static, please check this capture.

![image](https://user-images.githubusercontent.com/2866694/55280812-379a5280-52f0-11e9-8d5b-6e0d927b480e.png)

**Comment by sirbrillig** [comment]
Clearly you've found a bug! I'm having trouble reproducing it, though. Could you provide more of a context of your example? If I do this, it seem to not report an error:

```php
class ClassWithStaticFunction {
    function method_with_static_assigned_local_var() {
        static $new_var = null;
        if ($new_var === null) {
            echo 'it is null';
        }
    }
}
```

**Comment by Rolige** [comment]
Hi @sirbrillig,

The code that you leave does not give any error, apparently the problem occurs when the static variable is defined within a control structure. I could be able to generate the problem with the following code.

```php
<?php

class ClassWithStaticVarFunction
{
    public function method_with_static_assigned_local_var()
    {
        $var1 = 1;

        if ($var1 === 1) {
            static $new_var = null;

            if ($new_var === null) {
                echo 'it is null';
            }
        }
    }
}
```

Here an evidence.
![image](https://user-images.githubusercontent.com/2866694/55342255-9fbe7500-5465-11e9-8390-eecf53fca7b8.png)

Hope this help you to fix this.


Regards!

**Comment by sirbrillig** [comment]
Thank you! That was the hint I needed.

**Comment by Rolige** [comment]
Thanks, it works perfect now!
